app/models/currency.rb

```diff
@@ -0,0 +1,23 @@
+class Currency < ApplicationRecord
+  before_create :set_date
```

Comment:
a record with date as nil should not be return true when valid? is called on it

Reply:
fixed it with before_validation